opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -378,16 +383,9 @@ impl<E: PushMetricExporter> PeriodicReaderInner<E> {
         }
     }
 
-    fn collect_and_export(&self) -> OTelSdkResult {
-        // TODO: Reuse the internal vectors. Or refactor to avoid needing any
-        // owned data structures to be passed to exporters.
-        let mut rm = ResourceMetrics {
-            resource: Resource::empty(),
-            scope_metrics: Vec::new(),
-        };
-
+    fn collect_and_export(&self, rm: &mut ResourceMetrics) -> OTelSdkResult {
         let current_time = Instant::now();
-        let collect_result = self.collect(&mut rm);
+        let collect_result = self.collect(rm);
```

Comment:
I guess this still works without the need to clear the `scope_metrics` vec within the `ResourceMetrics`. I'm not sure if we have tests for this already but we need to ensure that clearing is not required by verifying that subsequent collects don't continue to show the old data from the previous collect (in case of Delta).